iceoryx_hoofs/include/iceoryx_hoofs/memory/relative_pointer.hpp

```diff
@@ -139,20 +130,20 @@ class RelativePointer
 
     /// @brief Get the base pointer associated with this' id
     /// @return The registered base pointer of the RelativePointer object
-    ptr_t getBasePtr() const noexcept;
+    T* getBasePtr() const noexcept;
```

Comment:
After discussing this topic with @MatthiasKillat we concluded that it's better to use `T*` here because otherwise due to `M3-9-1` we would have to write in the `.inl` `typename RelativePointer<T>::ptr_t` which is not really nice.

Reply:
Well, then we have an inconsistency since we use `typename RelativPointer<T>::offset_t` instead of just `std::uintptr_t`. With your argumentation we should also not use that.

Reply:
@elBoberido `T*` return was mainly for `consistency` of how the STL [smart pointers}(https://en.cppreference.com/w/cpp/memory/shared_ptr). This in particular defines no pointer type but only an element_type (which is T). The important point is to easily see that `T*` is the return type, as this is part of the contract.

Writing less is a side benefit.